lib/sentry/context.ex

```diff
@@ -40,26 +61,129 @@ defmodule Sentry.Context do
     }
   end
 
+  @doc """
+  Merges new fields into the `:extra` context, specific to the current process.
+
+  This is used to set fields which should display when looking at a specific
+  instance of an error.
+
+  ## Example
+
+      iex> Sentry.Context.set_extra_context(%{id: 123})
+      nil
+      iex> Sentry.Context.set_extra_context(%{detail: "bad_error"})
+      %{extra: %{id: 123}}
+      iex> Sentry.Context.set_extra_context(%{message: "Oh no"})
+      %{extra: %{detail: "bad_error", id: 123}}
+      iex> Sentry.Context.get_all()
+      %{
+        user: %{},
+        tags: %{},
+        extra: %{detail: "bad_error", id: 123, message: "Oh no"},
+        request: %{},
+        breadcrumbs: []
+      }
+  """
+  @spec set_extra_context(map()) :: nil | map()
   def set_extra_context(map) when is_map(map) do
     get_context()
     |> set_context(@extra_key, map)
   end
 
+  @doc """
+  Merges new fields into the `:user` context, specific to the current process.
+
+  This is used to set certain fields which identify the actor who experienced a
+  specific instance of an error.
+
+  ## Example
+
+      iex> Sentry.Context.set_user_context(%{id: 123})
+      nil
+      iex> Sentry.Context.set_user_context(%{username: "george"})
+      %{user: %{id: 123}}
+      iex> Sentry.Context.get_all()
+      %{
+        user: %{id: 123, username: "george"},
+        tags: %{},
+        extra: %{},
+        request: %{},
+        breadcrumbs: []
+      }
+  """
+  @spec set_user_context(map()) :: nil | map()
   def set_user_context(map) when is_map(map) do
     get_context()
     |> set_context(@user_key, map)
   end
 
+  @doc """
+  Merges new fields into the `:tags` context, specific to the current process.
+
+  This is used to set fields which should display when looking at a specific
+  instance of an error. These fields can also be used to search and filter on.
+
+  ## Example
+
+      iex> Sentry.Context.set_tags_context(%{id: 123})
+      nil
+      iex> Sentry.Context.set_tags_context(%{other_id: 456})
+      %{tags: %{id: 123}}
+      iex> Sentry.Context.get_all()
+      %{
+          breadcrumbs: [],
+          extra: %{},
+          request: %{},
+          tags: %{id: 123, other_id: 456},
+          user: %{}
+      }
+  """
+  @spec set_tags_context(map()) :: nil | map()
   def set_tags_context(map) when is_map(map) do
     get_context()
     |> set_context(@tags_key, map)
   end
 
+  @doc """
+  Merges new fields into the `:request` context, specific to the current
+  process.
+
+  This is used to set metadata that identifies the request associated with a
+  specific instance of an error.
```

Comment:
Ditto with this. I couldn't actually find documentation on what this even does in current versions of Sentry (it's possible I didn't look hard enough). Let me know if this is not accurate.